corehq/util/doc_processor/sql.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+
+from django.conf import settings
+
+from corehq.util.doc_processor.interface import DocumentProvider
+from corehq.util.pagination import ResumableFunctionIterator, ArgsProvider
+
+
+class SqlModelArgsProvider(ArgsProvider):
+    def __init__(self, model_filter_attribute):
+        self.model_filter_attribute = model_filter_attribute
+        if not settings.USE_PARTITIONED_DATABASE:
+            self.db_list = [None]  # use the default database
+        else:
+            from corehq.sql_db.config import PartitionConfig
+            partition_config = PartitionConfig()
+            self.db_list = partition_config.get_form_processing_dbs()
+
+    def get_initial_args(self):
+        return [self.db_list[0], datetime.min, None], {}
+
+    def get_next_args(self, result, *last_args, **last_view_kwargs):
+        if result:
+            next_filter_value = getattr(result, self.model_filter_attribute)
+            next_id = result.pk
+            return [last_args[0], next_filter_value, next_id], {}
+        else:
+            last_db = last_args[0]
+            # skip databases already processed
+            index = self.db_list.index(last_db) + 1
+            self.db_list = self.db_list[index:]
+            try:
+                next_db = self.db_list[0]
+            except IndexError:
+                raise StopIteration
+
+            return [next_db, datetime.min, None], {}
+
+
+def ResumableSqlModelIterator(iteration_key, reindex_accessor, chunk_size=100, event_handler=None):
```

Comment:
Does it make sense to rename this like 3893c34d7c50fd31cf087121ff03998799b1780e?